semantic-core/src/Analysis/Eval.hs

```diff
@@ -36,9 +37,17 @@ eval :: ( Carrier sig m
 eval Analysis{..} eval = \case
   Var n -> lookupEnv' n >>= deref' n
   Term c -> case c of
-    Let n -> alloc n >>= bind n >> unit
+    Rec (Named (Ignored n) b) -> do
+      addr <- alloc n
+      v <- bind n addr (eval (instantiate1 (pure n) b))
+      v <$ assign addr v
```

Comment:
`Rec` implements recursive bindings of the form `rec x = y` s.t. occurrences of `x` within `y` are bound to the result of `y`. It’s not strictly requisite given mutable variables, but it’s hella convenient, and easier to analyze than the equivalent mutation-based knot-tying.